components/renku_data_services/notebooks/api/classes/k8s_client.py

```diff
@@ -259,26 +234,17 @@ def patch_image_pull_secret(self, server_name: str, gitlab_token: GitlabToken) -
                 "value": base64.b64encode(json.dumps(new_docker_config).encode()).decode(),
             }
         ]
-        self._core_v1.patch_namespaced_secret(
-            secret_name,
-            self.namespace,
-            patch,
-        )
+        await secret.patch(patch, type="json")
 
-    def patch_statefulset_tokens(self, name: str, renku_tokens: RenkuTokens) -> None:
+    async def patch_statefulset_tokens(self, name: str, renku_tokens: RenkuTokens) -> None:
         """Patch the Renku and Gitlab access tokens that are used in the session statefulset."""
         try:
-            sts = self._apps_v1.read_namespaced_stateful_set(name, self.namespace)
-        except ApiException as err:
-            if err.status == 404:
-                # NOTE: It can happen potentially that another request or something else
-                # deleted the session as this request was going on, in this case we ignore
-                # the missing statefulset
-                return
-            raise
+            sts = cast(StatefulSet, await StatefulSet.get(name=name, namespace=self.namespace))
+        except NotFoundError:
+            return None
 
-        containers: list[V1Container] = sts.spec.template.spec.containers
-        init_containers: list[V1Container] = sts.spec.template.spec.init_containers
+        containers: list[V1Container] = [V1Container(**i) for i in sts.spec.template.spec.containers]
+        init_containers: list[V1Container] = [V1Container(**i) for i in sts.spec.template.spec.init_containers]
```

Comment:
nitpick: I wouldn't use `i` as a loop variable for something that isn't an index

Reply:
Agreed, I already changed a similar loop in another part.
Should this be done in this PR or left for the following cleanup PR ?

Reply:
i think small things like that should be done in this PR otherwise they'll get lost

Reply:
Done for all the loops within the notebook volume.